packages/client/components/ViewerNotOnTeam.tsx

```diff
@@ -41,10 +42,15 @@ const ViewerNotOnTeam = (props: Props) => {
           {invitationToken: teamInvitation.token},
           {history, meetingId}
         )
-      } else if (isOnTeam) {
-        // if already on the team, goto team dash
-        const redirectTo = getValidRedirectParam()
-        const nextRoute = redirectTo || `/team/${teamId}`
+        return
+      }
+      const redirectTo = getValidRedirectParam()
+      // if already on the team, go to team dash
+      const nextRoute = redirectTo || `/team/${teamId}`
+      const isRoutingToMeeting = nextRoute === `/meet/${meetingId}`
+      const isRoutingToTeam = nextRoute === `/team/${teamId}`
+      const isNextRoutePossible = (isOnTeam && isRoutingToTeam) || (meeting && isRoutingToMeeting)
```

Comment:
if the problem is with the auth token updating before the query updates, I think we can simply this:
```
if (teamInvitation) acceptTeamInvitation...
else if (meeting) history.replace(/meet/meetingId)
else if (team) history.replace('/team/teamId')
else PushInvitation

Reply:
makes sense! I've removed the `authObj` check here and in the TeamContainer